File: built-in plugins dsp/mb-vocal-exciter.cpp
/**
 * MB Vocal Exciter
 * Category : effect
 * Type     : vocal
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Vocal clarity enhancer
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_VOCAL_EXCITER_H
#define MB_VOCAL_EXCITER_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbVocalExciter : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-vocal-exciter";
    static constexpr const char* PLUGIN_NAME    = "MB Vocal Exciter";
    static constexpr const char* PLUGIN_TYPE    = "vocal";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float amount = 0.3f;  // range [0, 1]
    float freq = 3000f;  // range [1000, 8000]
    float mix = 0.5f;  // range [0, 1]
    };

    MbVocalExciter() = default;
    ~MbVocalExciter() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.amount = std::clamp(params.amount, 0f, 1f);
        params.freq = std::clamp(params.freq, 1000f, 8000f);
        params.mix = std::clamp(params.mix, 0f, 1f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Vocal Exciter
        return input;
    }
};

#endif // MB_VOCAL_EXCITER_H

File: built-in plugins dsp/mb-vocal-exciter.ts
import type { PluginDefinition } from "../server/services/pluginHostService";

const MbVocalExciterPlugin: PluginDefinition = {
  id: "mb-vocal-exciter",
  slug: "mb-vocal-exciter",
  name: "MB Vocal Exciter",
  category: "effect",
  type: "vocal",
  version: "1.0.0",
  description: "Vocal clarity enhancer",
  author: "Max Booster",
  grade: "A",
  parameters: [
    {
      id: "amount",
      name: "Amount",
      type: "float",
      defaultValue: 0.3,
      minValue: 0,
      maxValue: 1,
      automatable: true,
    },
    {
      id: "freq",
      name: "Frequency",
      type: "float",
      defaultValue: 3000,
      minValue: 1000,
      maxValue: 8000,
      automatable: true,
    },
    {
      id: "mix",
      name: "Mix",
      type: "float",
      defaultValue: 0.5,
      minValue: 0,
      maxValue: 1,
      automatable: true,
    },
  ],
  defaultPreset: { amount: 0.3, freq: 3000, mix: 0.5 },
};

export default MbVocalExciterPlugin;
